plugins/catalog-backend-module-ldap/src/ldap/client.ts

```diff
@@ -147,39 +147,53 @@ export class LdapClient {
   async searchStreaming(
     dn: string,
     options: SearchOptions,
-    f: (entry: SearchEntry) => void,
+    f: (entry: SearchEntry) => Promise<void> | void,
   ): Promise<void> {
     try {
       return await new Promise<void>((resolve, reject) => {
         // Note that we clone the (frozen) options, since ldapjs rudely tries to
         // overwrite parts of them
-        this.client.search(dn, cloneDeep(options), (err, res) => {
-          if (err) {
-            reject(new Error(errorString(err)));
-          }
-
-          res.on('searchReference', () => {
-            this.logger.warn('Received unsupported search referral');
-          });
-
-          res.on('searchEntry', entry => {
-            f(entry);
-          });
-
-          res.on('error', e => {
-            reject(new Error(errorString(e)));
-          });
-
-          res.on('end', r => {
-            if (!r) {
-              throw new Error('Null response');
-            } else if (r.status !== 0) {
-              throw new Error(`Got status ${r.status}: ${r.errorMessage}`);
-            } else {
-              resolve();
+        this.client.search(
+          dn,
+          merge(cloneDeep(options), { paged: { pagePause: true } }),
```

Comment:
I just realized that this will mean that you'll forcibly enable pagination, no matter what the user had defined beforehand. I think you can't do that unfortunately; there are LDAP servers that just do not support pagination at all, and those will fail if we merge this. The code must function both in paged and non-paged circumstances.

Maybe you could make a new helper function, something like this

```ts
function createOptions(inputOptions: SearchOptions): SearchOptions {
  const result = cloneDeep(options);

  if (result.paged === true) {
    result.paged = { pagePause: true };
  } else if (typeof result.paged === 'object') {
    result.paged.pagePause = true;
  }

  return result;
}
```